app/javascript/app/greetingSlice.js

```diff
@@ -0,0 +1,49 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+
+const initialState = {
+  greeting: {},
+  isLoading: false,
+  isError: false
+};
+
+const URL = 'http://localhost:3000/api/greetings';
+
```

Comment:
🚀 Great job setting up your project. Everything is configured well in your project 💯 

However, there is an issue with displaying the greeting on the page. The API endpoint is set up correctly and returns a random greeting as expected. The issue seems to be with CORS preventing the HTTP request from redux.

![image](https://github.com/Jeddaa/my_new_rails_app/assets/38088818/644ec000-a82a-4d8f-ae79-d6dec41f1c91)

Here's one way you can resolve this, although you are free to find another solution if this one doesn't work for you.

1. The easiest way to configure CORS on your Rails app is to use [rack-cors gem](https://github.com/cyu/rack-cors). You can install it by adding the following line to your Gemfile: 

```ruby
gem 'rack-cors'
```

2. Next, you need to provide the configuration for the gem. You need to inform Rails which origin it should allow. To do that, you need to create a new initializer for your application. Create the `cors.rb` file inside the `config/initializers` directory.

The content of the `config/initializers/cors.rb` should be similar to this: 

```ruby
Rails.application.config.middleware.insert_before 0, Rack::Cors do
  allow do
    origins 'http://example.com:80' #modify the origin to be the URL of your API endpoint 
    resource '*', headers: :any, methods: [:get, :post]
  end
end
```

Reply:
thank you for catching that as i didn't have that issue